src/core/server.py

```
@@ -232,21 +270,100 @@ async def crawl4ai_lifespan(server: FastMCP) -> AsyncIterator[Crawl4AIContext]:
 mcp = FastMCP(
     "mcp-crawl4ai-rag",
     lifespan=crawl4ai_lifespan,
-    middleware=[TimingMiddleware(), CancellationHandlingMiddleware()],
+    middleware=[TimingMiddleware(), TimeoutMiddleware(), CancellationHandlingMiddleware()],
 )
 
 # CancellationHandlingMiddleware now handles cancellation gracefully without response conflicts
 
 async def health_check(request):
```

Ruff (0.12.2)

Missing return type annotation for public function `health_check`

(ANN201)